src/Functional/Curry.php

```diff
@@ -35,14 +36,18 @@
  */
 function curry(callable $function, $required = true)
 {
-    if (is_string($function) && strpos($function, '::', 1) !== false) {
-        $reflection = new ReflectionMethod($function);
-    } elseif (is_array($function) && count($function) === 2) {
-        $reflection = new ReflectionMethod($function[0], $function[1]);
-    } elseif (is_object($function) && method_exists($function, '__invoke')) {
-        $reflection = new ReflectionMethod($function, '__invoke');
+    if(method_exists('\Closure','fromCallable')) {
```

Comment:
missing space after `if` and just `'Closure'`, without the trailing namespace separator should work.